File: engine/tests/fixtures/phase4_discovery/typescript/full_surface.ts
import OpenAI from "openai";

const client = new OpenAI({ maxRetries: 3 });
const alias = client;
const responses = alias.responses;

export async function declaredWrapper(signal: AbortSignal) {
  try {
    return await responses.create({
      model: "gpt-5",
      input: "fixture",
      stream: true,
      tools: [],
      text: { format: { type: "json_schema" } },
      signal,
    });
  } catch (error) {
    throw error;
  }
}

File: engine/tests/fixtures/phase4_discovery/python/full_surface.py
import asyncio
import openai
from openai import AsyncOpenAI

client = AsyncOpenAI(max_retries=3)
responses = client.responses


async def declared_wrapper() -> object:
    try:
        return await responses.create(
            model="gpt-5",
            input="fixture",
            stream=True,
            tools=[],
            text={"format": {"type": "json_schema"}},
            timeout=10,
        )
    except (openai.APIError, asyncio.CancelledError):
        raise
